pep-9999.rst

```diff
@@ -0,0 +1,507 @@
+
+* pip install app
+* pip install app --no-index --find-links https://example.com/
+* pip install --editable "git+https://example.com/repo/app.git#egg=app&subdirectory=setup"
+* pip install -e ./app
+
+Use cases
+=========
+
+"Freezing" an environment
+
+  Tools, such as ``pip freeze``, which generate requirements from the Database
+  of Installed Python Distributions SHOULD exploit ``direct_url.json``
+  if it is present, and give it priority over the Version metadata in order
+  to generate a higher fidelity output. In the presence of a ``vcs`` direct URL reference,
+  the ``commit_id`` field SHOULD be used in priority in order to provide
+  the highest possible fidelity to the originally installed version. If
+  supported by their requirement format, tools are encouraged also to output
+  the ``tag`` value if present, as it has immutable semantics.
+  Tools MAY choose another approach, depending on the needs of their users.
+
+Backwards Compatibility
+=======================
+
+Since this PEP specifies a new file in the ``.dist-info`` directory,
+there are no backwards compatibility implications.
+
+Alternatives
+============
+
+PEP426 source_url
+-----------------
+
+The now withdrawn PEP 426 specifies a ``source_url`` metadata entry.
+It is also implemented in `distlib`_.
+
+It was intended for a slightly different purpose, for use in sdists.
+
+This format lacks support for the ``subdirectory`` option of pip requirement
+URLs. The same limitation is present in `PEP440 Direct References`_.
+
+It also lacks explicit support for `environment variables in the user:password
+part of URLs`_.
+
+The introduction of a key/value extensibility mechanism and support
+for environment variables for user:password in PEP 440, would be necessary
+for use in this PEP.
+
+revision vs ref
+---------------
+
+The ``requested_revision`` key was retained over ``requested_ref`` as it is a more generic term
+across various VCS and ``ref`` has a specific meaning for ``git``.
+
+
+References
+==========
+
+.. _`pip issue #609`: https://github.com/pypa/pip/issues/609
+.. _`thread on discuss.python.org`:  https://discuss.python.org/t/pip-freeze-vcs-urls-and-pep-517-feat-editable-installs/1473
+.. _PEP440: http://www.python.org/dev/peps/pep-0440
+.. _`VCS URLs supported by pip`: https://pip.pypa.io/en/stable/reference/pip_install/#vcs-support
+.. _`PEP440 Direct References`: https://www.python.org/dev/peps/pep-0440/#direct-references
+.. _`Pipfile and Pipfile.lock`: https://github.com/pypa/pipfile
+.. _distlib: https://distlib.readthedocs.io
+.. _`environment variables in the user:password part of URLs`: https://pip.pypa.io/en/stable/reference/pip_install/#id10
+
+Acknowledgements
+================
+
+Various people helped make this PEP a reality. Paul F. Moore provided the
+essence of the abstract. Nick Coghlan suggested the direct_url name.
+
+Copyright
+=========
+
+This document has been placed in the public domain.
```

Comment:
Hello @sbidoul,

The requirements at [PEP 1](https://www.python.org/dev/peps/pep-0001/#what-belongs-in-a-successful-pep) were changed recently by #1117, so can you please update the `Copyright` section to make this PEP meet them?
> 13. Copyright/license -- Each new PEP must be placed under a dual license of public domain and CC0-1.0-Universal [8] (see this PEP for an example).

```suggestion
This document is placed in the public domain or under the CC0-1.0-Universal license, whichever is more permissive.
```